journal.py

```diff
@@ -72,9 +73,51 @@ def list_view(request):
     return {'entries': entries}
 
 
+@view_config(route_name='detail', renderer='templates/detail.jinja2')
+def detail_view(request, session=None):
+    entry_id = request.params.get('id')
+    if session is None:
+        session = DBSession
+    entry = session.query(Entry).get(entry_id)
+    entry.content_md = markdown.markdown(
+        entry.content,
+        extensions=['markdown.extensions.codehilite'],
+        extension_configs={'markdown.extensions.codehilite': {'noclasses': True}}
+    )
+    return {'entry': entry}
+
+
+@view_config(route_name='edit', renderer='templates/edit.jinja2')
+def edit_view(request, session=None):
+    if not request.authenticated_userid:               # hackers get redirected
+        return HTTPFound(request.route_url('login'))   # to login
+    entry_id = request.params.get('id')
+    if session is None:
+        session = DBSession
```

Comment:
This shim should not be required here.  Testing the views should generally take place inside a valid session, so that the request is something.  That means that DBSession should be safe to use in this scope without need to pass in a session alternately.